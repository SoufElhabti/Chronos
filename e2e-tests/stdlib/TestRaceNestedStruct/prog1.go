package main

func main() {
	type X struct {
		x, y int
	}
	type Y struct {
		x X
	}
	c := make(chan Y)
	var y Y
	go func() {
		c <- y
	}()
	y.x.y = 42
	<-c
}
