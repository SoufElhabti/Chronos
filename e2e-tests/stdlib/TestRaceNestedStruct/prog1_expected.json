{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:2",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:2",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:13:3",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 3,
					"2": 1
				}
			}
		},
		{
			"ID": 4,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:13:8",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:11:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 3,
					"2": 2
				}
			}
		},
		{
			"ID": 5,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:4",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:4",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 6,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:6",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 5
				}
			}
		},
		{
			"ID": 7,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:6",
			"Value": "-",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 8,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:6",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:16:6",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 9,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:11",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:10:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		},
		{
			"ID": 10,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:9",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				}
			}
		},
		{
			"ID": 11,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:2",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:9",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 10
				}
			}
		},
		{
			"ID": 12,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:2",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 11
				}
			}
		},
		{
			"ID": 13,
			"Pos": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:18:6",
			"Value": "/Users/amdavidson/dev/go/src/github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceNestedStruct/prog1.go:17:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 12
				}
			}
		}
	]
}